sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1618,39 +1613,28 @@ case class ArrayBinarySearch(array: Expression, value: Expression)
     override def compare(o1: Any, o2: Any): Int =
       (o1, o2) match {
         case (null, null) => 0
-        case (null, _) => 1
```

Comment:
fix bug

Reply:
do you mean existing implementation has a bug?

Reply:
Yes, it's a bug from a previous implementation.

Reply:
Which versions of Spark are affected by this bug Would it be better to submit a separate PR to fix this bug first?

Reply:
Same question here. Is it a new regression or a long standing bug?

Reply:
It is a new bug. I have already privately replied to @LuciferYang  and forgot to reply on the PR. I am very sorry 😄